src/lib/project-saver-hoc.jsx

```diff
@@ -44,9 +44,17 @@ const ProjectSaverHOC = function (WrappedComponent) {
         constructor (props) {
             super(props);
             bindAll(this, [
+                'leavePageConfirm',
                 'tryToAutoSave'
             ]);
         }
+        componentWillMount () {
+            if (typeof window === 'object') {
+                // Note: it might be better to use a listener instead of assigning onbeforeunload;
+                // but then it'd be hard to turn this listening off in our tests
+                window.onbeforeunload = () => this.leavePageConfirm;
```

Comment:
This code looks like it is just returning the function `this.leavePageConfirm`, rather than returning the result of `this.leavePageConfirm(event)`. This will result in the alert always showing–[MDN says](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload#Notes):
> When this event returns (or sets the returnValue property to) a value other than null or undefined, the user is prompted to confirm the page unload.